"no viable conversion from numpy_expr to numpy_gexpr"

In the latest version of pythran, on mac os, this does not compile
```
#pythran export test(float64[], int, int)
def fancy(signal, k):
    sub = signal[1:]
    s = sub[0]
    if k:
        sub = sub[1:]
    return sub[0] + s

def test(a, cutoff, k):
    return fancy(-a[0:cutoff], k)
```
The error is
```
/var/folders/2p/dngs2w5x6tn3lw2jqsnl42_80000gp/T/tmp_at0sqcs.cpp:95:15: error: no viable conversion from 'decltype((*this)._index((s.lower , s), std::make_index_sequence<sizeof...(Args)>{}))' (aka '(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>>') to '__type8' (aka 'numpy_gexpr<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>')
      __type8 sub = pythonic::types::as_const(signal)[pythonic::types::contiguous_slice(1L,pythonic::builtins::None)];
              ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/assignable.hpp:44:40: note: in instantiation of function template specialization '(anonymous namespace)::__pythran_bug::fancy::operator()<(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>> &, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>>, long>' requested here
    return std::forward<F>(f).template operator()<by_val_t<Args>...>(
                                       ^
/var/folders/2p/dngs2w5x6tn3lw2jqsnl42_80000gp/T/tmp_at0sqcs.cpp:107:31: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::call<(anonymous namespace)::__pythran_bug::fancy, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>> &, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>>, long &>' requested here
      return pythonic::types::call(fancy(), pythonic::operator_::neg(pythonic::types::as_const(a)[pythonic::types::contiguous_slice(0L,cutoff)]), k);
                              ^
/var/folders/2p/dngs2w5x6tn3lw2jqsnl42_80000gp/T/tmp_at0sqcs.cpp:119:65: note: in instantiation of function template specialization '(anonymous namespace)::__pythran_bug::test::operator()<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, long, long>' requested here
                                auto res = __pythran_bug::test()(a, cutoff, k);
                                                                ^
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/numpy_gexpr.hpp:594:5: note: candidate constructor not viable: no known conversion from 'decltype((*this)._index((s.lower , s), std::make_index_sequence<sizeof...(Args)>{}))' (aka '(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>>') to 'const numpy_gexpr<ndarray<double, pshape<long>>, cstride_normalized_slice<1>> &' for 1st argument
    numpy_gexpr(numpy_gexpr const &) = default;
    ^
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/numpy_gexpr.hpp:595:5: note: candidate constructor not viable: no known conversion from 'decltype((*this)._index((s.lower , s), std::make_index_sequence<sizeof...(Args)>{}))' (aka '(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long>>, (anonymous namespace)::pythonic::types::cstride_normalized_slice<1>>>') to 'numpy_gexpr<ndarray<double, pshape<long>>, cstride_normalized_slice<1>> &&' for 1st argument
    numpy_gexpr(numpy_gexpr &&) = default;
    ^
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/numpy_gexpr.hpp:599:5: note: candidate template ignored: could not match 'numpy_gexpr' against 'numpy_expr'
    numpy_gexpr(numpy_gexpr<Argp, S...> const &other);
```

--- a/pythran/pythonic/include/types/numpy_expr.hpp
+++ b/pythran/pythonic/include/types/numpy_expr.hpp
@@ -865,5 +865,12 @@ struct __combined<pythonic::types::numpy_texpr<T>, pythonic::types::numpy_expr<O
       pythonic::types::array_tuple<long, pythonic::types::numpy_expr<Op, Args...>::value>>;
 };
 
+template <class E, class... S, class Op, class... Args>
+struct __combined<pythonic::types::numpy_expr<Op, Args...>, pythonic::types::numpy_gexpr<E, S...>> {
+  using type = pythonic::types::ndarray<
+      typename pythonic::types::numpy_expr<Op, Args...>::dtype,
+      pythonic::types::array_tuple<long, pythonic::types::numpy_expr<Op, Args...>::value>>;
+};
+
 /*}*/
 #endif

--- a/pythran/tests/test_ndarray.py
+++ b/pythran/tests/test_ndarray.py
@@ -1659,3 +1659,18 @@ def test1(X,A):
         X = np.concatenate((np.zeros((N)),X))
     return X'''
         self.run_test(code, numpy.ones((10)), test_combiner_2=[NDArray[float,:]])
+
+    def test_combiner_3(self):
+        code = '''
+def fancy(signal, k):
+    sub = signal[1:]
+    s = sub[0]
+    if k:
+        sub = sub[1:]
+    return sub[0] + s
+
+def test_combiner_3(a, cutoff, k):
+    return fancy(-a[0:cutoff], k)
+'''
+        self.run_test(code, numpy.ones((10)), 5, 1,
+                      test_combiner_3=[NDArray[float,:], int, int])

